### PR TITLE
fix: #62203 - Add max, min props for Date Picker

### PR DESCRIPTION
**Pull request by herda1**
[AB#62203 ](https://dev.azure.com/EquinorASA/5f193d29-b7e1-4e23-aba0-7eac7e847ab5/_workitems/edit/62203 )

## Prerequisites

Please make sure you can check the following boxes:

-   [x] My code follows the code style of this project
-   [x] All new and existing tests passed
-   [x] Fix any eslint/prettier warnings/errors: npm run lint
-   [x] Remove unnecessary console.log (use logInfo logWarn if it's needed)
-   [x] Are variable and function names properly describing what it does?

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

-   [x] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
-   [ ] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] I have updated the documentation accordingly
-   [ ] I have added tests to cover my changes

### Description

- Decided to use maxDate and minDate from ReactDatePicker, by which we will ensure validation of dates in EchopediaWeb. 

- Tweakings on the icon style: 
  iconHover

### Remarks

- Increase EchoComponents version and use it in EchopediaWeb: will do once this PR is approved

### CODE REVIEW

File: .storybook/stories/components/datePicker/DatePickerWrapper.tsx [hunk 1]
```diff
@@ -5,16 +5,17 @@ import TimePicker from '../../../../src/components/timePicker/TimePicker';
 const ReactDatePickerWrapper: React.FC<DatePickerProps> = ({
     id,
     value,
-    label,
```
Comment (herda1) [inline .storybook/stories/components/datePicker/DatePickerWrapper.tsx:8]: this value was unused

File: src/components/datePicker/ReactDatePicker.tsx [hunk 2]
```diff
@@ -20,23 +37,10 @@ export interface DatePickerProps {
...
-        | 'right-start'
-        | 'right'
-        | 'right-end'
-        | 'bottom-end'
-        | 'bottom'
-        | 'bottom-start'
-        | 'left-end'
-        | 'left'
-        | 'left-start';
+    popperPlacement?: PopperPlacement;
```
Comment (herda1) [inline src/components/datePicker/ReactDatePicker.tsx:40]: extracted into a type object. Feel free to suggest for different names

File: src/components/datePicker/datePicker.module.css [hunk 3]
```diff
@@ -19,7 +19,6 @@
 
 .dateInput {
     background-color: rgba(247, 247, 247, 1) !important;
-    border-bottom: none !important;
```
Comment (herda1) [inline src/components/datePicker/datePicker.module.css:22]: this property was igonored, no affect

File: src/components/datePicker/ReactDatePicker.tsx [hunk 4]
```diff
@@ -48,7 +52,9 @@ export const ReactDatePicker: React.FC<DatePickerProps> = ({
     disableBeforeDate,
     className,
     popperPlacement,
-    showTimeSelect
+    showTimeSelect,
+    maxDate,
+    minDate
```
Comment (herda1) [inline src/components/datePicker/ReactDatePicker.tsx:57]: maxDate and minDate props from the React Date Picker: https://github.com/Hacker0x01/react-datepicker/blob/master/docs/datepicker.md

File: src/components/datePicker/ReactDatePicker.tsx [hunk 5]
```diff
@@ -20,23 +37,10 @@ export interface DatePickerProps {
...
-        | 'bottom-end'
-        | 'bottom'
-        | 'bottom-start'
-        | 'left-end'
-        | 'left'
-        | 'left-start';
+    popperPlacement?: PopperPlacement;
     showTimeSelect?: boolean;
+    maxDate?: Date;
+    minDate?: Date;
```
Comment (herda1) [inline src/components/datePicker/ReactDatePicker.tsx:43]: optional maxDate and minDate props from the React Date Picker: https://github.com/Hacker0x01/react-datepicker/blob/master/docs/datepicker.md

File: src/components/datePicker/datePicker.module.css [hunk 6]
```diff
@@ -38,19 +37,17 @@
...
     height: 20px;
     color: #616161;
     cursor: pointer;
 }
 
 .clearDateValue {
-    position: absolute !important;
-    right: -4px;
-    top: 8px;
+    left:220px;
```
Comment (CsabaTheKiss) [inline src/components/datePicker/datePicker.module.css:48]: Missing space before 220

File: src/components/datePicker/datePicker.module.css [hunk 7]
```diff
@@ -38,19 +37,17 @@
...
+    bottom: 35px;
     cursor: pointer;
+    height:30px !important;
```
Comment (CsabaTheKiss) [inline src/components/datePicker/datePicker.module.css:51]: Missing space before 30

**Comment by CsabaTheKiss** [comment]
Hmmm, it seems the clock icon get's misplaced when clearing out the date input.

clock icon misplaced

File: .storybook/stories/components/datePicker/DatePickerWrapper.tsx (hunk 1 above)
Comment (khollund) [reply .storybook/stories/components/datePicker/DatePickerWrapper.tsx:8]: Is it possible to make it optional and pass it to the title if it's sent in? They seem to be designed without label in Echo sometimes, but it's a requirement in many of the standards we might want to follow.

**Comment by CsabaTheKiss** [comment]
@herda1 CSS fix for the icon looks good! 👍 It's ok from my side, but Kristian also left a comment regarding the removed label:
https://github.com/equinor/EchoComponents/pull/74#discussion_r835195744

File: .storybook/stories/components/datePicker/DatePickerWrapper.tsx (hunk 1 above)
Comment (khollund) [reply .storybook/stories/components/datePicker/DatePickerWrapper.tsx:8]: I didn't notice it was only for the wrapper, so not important unless we want it for the example 😁

**Review by CsabaTheKiss — APPROVED** [review]

**Pull request merged**